File: homework2/raodm_ex4.cpp
#include <algorithm>
#include <fstream>
#include <iostream>
#include <string>
#include <locale>
#include <iterator>

int main(int argc, char *argv[]) {
    if (argc < 2) {
        std::cout << "Specify <File> <toggle_char> ...\n";
        return 1;
    }
    // Read the file.
    std::ifstream dataFile(argv[1]);
    dataFile >> std::noskipws;
    std::istream_iterator<char> dataIter(dataFile), eof;
    std::string data(dataIter, eof);
    // Replace specified charachters
    for(int curr = 2; (curr < argc); curr++) {
        const char low = argv[curr][0];
        const char up  = toupper(low);
        auto  toggle = [low,up](char c) {
            return (c == low ? up : c == up ? low : c);
        };
        std::transform(data.begin(), data.end(), data.begin(), toggle);
    }
    // Print the modifed data.
    std::cout << data;
    return 0;
}

File: homework2/raodm_ex4.java
import java.nio.file.Files;
import java.nio.file.Paths;
import java.nio.charset.StandardCharsets;
import java.io.IOException;
import java.lang.Character;

class raodm_ex4 {
    static String toggleCase(String str, final char low) {
        final char up = Character.toUpperCase(low);
        char[] data   = str.toCharArray();
        for(int i = 0; (i < data.length); i++) {
            final char c = data[i];
            data[i] = (c == low ? up : c == up ? low : c);
        }
        return new String(data);
    }
    
    public static void main(String args[]) throws IOException {
        if (args.length < 2) {
            System.out.println("Specify <File> <toggle_char> ...");
            System.exit(1);
        }
        // Load all the bytes into a string using Java's non-blocking
        // I/O package (same as Python's read() method). However, it
        // is a bit of a cheating because it is much faster than the
        // iterator-based solution in C++.
        byte[] rawData = Files.readAllBytes(Paths.get(args[0]));
        String data    = new String(rawData, StandardCharsets.US_ASCII);
        // Replace specified charachters
        for(int curr = 1; (curr < args.length); curr++) {
            data = toggleCase(data, args[curr].charAt(0));
        }
        // Print the modifed data.
        System.out.print(data);
    }
}
